src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -210,12 +217,16 @@ public void validateUserCanRead(List<String> sourcePaths, AuthenticatedUserReque
       return;
     }
     // Obtain a token for the user's pet service account that can verify that it is allowed to read
-    String token;
-    try {
-      token = iamClient.getPetToken(user, GCS_VERIFICATION_SCOPES);
-    } catch (InterruptedException e) {
-      throw new PdaoException("Error obtaining a pet service account token");
-    }
+    String token =
```

Comment:
Is this doing something different than `AuthService.getPetAccountAuthToken()`? That exists and has a cache already.

Reply:
It looks like the scopes are different; either way it might be best to put this in `AuthService` since that's where we already have code that caches credentials.

Reply:
That's test code

Reply:
Right, but the same holds that auth service is test code so we can't access from here

Reply:
Yeah, I missed that this was in test code.

Abstraction-wise, I think it would be better to cache inside the `SamIam` code instead of doing it in the caller.